portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase

```diff
@@ -0,0 +1,206 @@
+			category = "Configuration",
+			panel = "Control Panel",
+			portlet = "Instance Settings Settings");
+
+		SystemSettings.gotoConfiguration(
+			configurationCategory = "SSO",
+			configurationName = "OpenID Connect",
+			configurationScope = "Virtual Instance Scope");
+
+		AssertNotVisible(locator1 = "OpenIDConnect#TOKEN_REFRESH_OFFSET");
+	}
+
+	@description = "This is a use cae for LPS-124900 TC-4: Verify that user session is terminated after a new access token is generated when the Refresh Token Offset is higher. "
+	@priority = "5"
+	test AssertUserIsLoggedOutWithHigherRefreshToken {
+		property keycloak.enabled = "true";
+		property osgi.module.configuration.file.names = "com.liferay.portal.security.sso.openid.connect.internal.configuration.OpenIdConnectProviderConfiguration~default.config:com.liferay.portal.security.sso.openid.connect.configuration.OpenIdConnectConfiguration.config";
+		property osgi.module.configurations = ":enabled=B&quot;true&quot;${line.separator}tokenRefreshOffset=I&quot;30&quot;";
+		property portal.upstream = "marcell";
+		property test.name.skip.portal.instance = "OpenIDConnect#AssertUserIsStillLoggedInAfterAccessTokenExpiresAndRenew";
+
+		OpenIDConnect.copyOSGIConnectionConfiguration(OSGiConfigFileName = "com.liferay.portal.security.sso.openid.connect.internal.configuration.OpenIdConnectProviderConfiguration_default.config");
+
+		ApplicationsMenu.gotoPortlet(
+			category = "Configuration",
+			panel = "Control Panel",
+			portlet = "System Settings");
+
+		SystemSettings.gotoConfiguration(
+			configurationCategory = "SSO",
+			configurationName = "OpenID Connect",
+			configurationScope = "System Scope");
+
+		Type(
+			locator1 = "OpenIDConnect#TOKEN_REFRESH_OFFSET",
+			value1 = "75");
+
+		AssertClick(
+			locator1 = "Button#SUBMIT",
+			value1 = "Save");
+
+		AssertTextEquals(
+			locator1 = "OpenIDConnect#TOKEN_REFRESH_OFFSET",
+			value1 = "75");
+
+		User.logoutPG();
+
+		User.loginUserWithOpenIDConnectEnabledPG(
+			userEmailAddress = "user1",
+			userPassword = "test");
+
+		Pause(locator1 = "65000");
+
+		User.viewLoggedOutPG();
+	}
+
+	@description = "This is a use case for LPS-124898 TC-1: Verify Instance admin can set the token refresh offset via System settings."
+	test TokenRefreshOffsetCanBeSetAtSystemLevel {
+		property portal.upstream = "marcell";
+		property test.name.skip.portal.instance = "OpenIDConnect#TokenRefreshOffsetCanBeSetAtSystemLevel";
+
+		ApplicationsMenu.gotoPortlet(
+			category = "Configuration",
+			panel = "Control Panel",
+			portlet = "System Settings");
+
+		SystemSettings.gotoConfiguration(
+			configurationCategory = "SSO",
+			configurationName = "OpenID Connect",
+			configurationScope = "System Scope");
+
+		AssertTextEquals.assertPartialText(
+			locator1 = "Message#INFO",
+			value1 = "This configuration is not saved yet. The values shown are the default.");
+
+		Type(
+			locator1 = "OpenIDConnect#TOKEN_REFRESH_OFFSET",
+			value1 = "60");
```

Comment:
Test with a different value than the default or verify the default is present before typing.